adcc/workflow.py

```diff
@@ -495,3 +509,73 @@ def setup_solver_printing(solmethod_name, matrix, kind, default_print,
         def inner_callback(state, identifier):
             default_print(state, identifier, output)
         return inner_callback
+
+
+def setup_environment(matrix, environment):
+    """
+    Setup environment matrix terms and/or energy corrections.
+    Internal function called from run_adc.
+    """
+    valid_envs = ["ptss", "ptlr", "linear_response"]
+    hf = matrix.reference_state
+    if hf.environment and environment is None:
+        raise InputError(
+            "Environment found in reference state, but no environment"
+            " configuration specified. Please select from the following"
+            f" schemes: {valid_envs}."
+        )
+    elif environment and not hf.environment:
+        raise InputError(
+            "Environment specified, but no environment"
+            " was found in reference state."
+        )
+    elif not hf.environment:
+        environment = False
+
+    if isinstance(environment, bool):
+        environment = {"ptss": True, "ptlr": True} if environment else {}
+    elif isinstance(environment, list):
+        environment = {k: True for k in environment}
+    elif isinstance(environment, str):
+        environment = {environment: True}
+    elif not isinstance(environment, dict):
```

Comment:
I think I would allow that. Not document it, but allow it. It makes your life easier for normalising the parameters.

Reply:
What exactly do you mean?

Reply:
To avoid exactly what you did here ... this statement causes a bug if  hf.environment is false, because than line 533 makes it a dict. I think if you remove this branch and allows dicts it would allow you to simplify the parsing logic.

Reply:
Dicts are of course allowed, but at this point, all other types are converted to a dict. The branch is just to catch types that we don't convert to a dict.
Sorry, I'm confused 😞

Reply:
I think it's again me (sorry I have a really busy week): I overlooked the `not`. All good ... in that case don't you think that the parsing logic can be made a little simpler? Maybe you can try to reduce on the number of branches / the additional on-line if ... then ... else thingy

Reply:
I have restructured this part a bit and made it a bit more pythonic 😄